### ISSUE TITLE
Drawings panel: regex filters can cause exceptions

### ISSUE THREAD

**Issue by Gama11** [magmaOffenburg/RoboViz#52]
- replace the filter text by `*`
- press the regex button
- run `RVTester` (seems like there need to be drawings)

```
Caused by: java.util.regex.PatternSyntaxException: Dangling meta character '*' near index 0
*
^
    at java.util.regex.Pattern.error(Pattern.java:1955)
    at java.util.regex.Pattern.sequence(Pattern.java:2123)
    at java.util.regex.Pattern.expr(Pattern.java:1996)
    at java.util.regex.Pattern.compile(Pattern.java:1696)
    at java.util.regex.Pattern.<init>(Pattern.java:1351)
    at java.util.regex.Pattern.compile(Pattern.java:1028)
    at java.util.regex.Pattern.matches(Pattern.java:1133)
    at java.lang.String.matches(String.java:2108)
    at rv.ui.DrawingListPanel.setListChanged(DrawingListPanel.java:194)
    at rv.comm.drawing.Drawings.fireShapeChangeListener(Drawings.java:96)
    at rv.comm.drawing.Drawings.update(Drawings.java:204)
    at rv.Viewer.update(Viewer.java:379)
    at js.jogl.prog.GLProgram.display(GLProgram.java:72)
    at jogamp.opengl.GLDrawableHelper.display(GLDrawableHelper.java:191)
    at javax.media.opengl.awt.GLCanvas$DisplayAction.run(GLCanvas.java:873)
    at jogamp.opengl.GLDrawableHelper.invokeGL(GLDrawableHelper.java:363)
    at javax.media.opengl.awt.GLCanvas$DisplayOnEventDispatchThreadAction.run(GLCanvas.java:890)
    at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:301)
    at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:744)
    at java.awt.EventQueue.access$400(EventQueue.java:97)
    at java.awt.EventQueue$3.run(EventQueue.java:697)
    at java.awt.EventQueue$3.run(EventQueue.java:691)
    at java.security.AccessController.doPrivileged(Native Method)
    at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:75)
    at java.awt.EventQueue.dispatchEvent(EventQueue.java:714)
    at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
    at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
    at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
    at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
    at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
    at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
```

Invalid regexes shouldn't be able to crash RoboViz. Ideally, the regex button is greyed out if the regex in the textfield is invalid, and have a tooltip that indicates this.

**Comment by hannesbraun** [comment]
Fixed with acc9cf149e892a2b7f2d15ee34ddca36b2e3d909

**Comment by hannesbraun** [comment]
Since there's also the possibility to press enter to apply the regex, I think it's more consistent to just treat an invalid regex as "match everything".
